lib/mpl_toolkits/mplot3d/proj3d.py

```diff
@@ -173,19 +173,20 @@ def _ortho_transformation(zfront, zback):
 def _proj_transform_vec(vec, M):
     vecw = np.dot(M, vec)
     w = vecw[3]
-    # clip here..
     txs, tys, tzs = vecw[0]/w, vecw[1]/w, vecw[2]/w
     return txs, tys, tzs
 
 
-def _proj_transform_vec_clip(vec, M):
+def _proj_transform_vec_clip(vec, M, focal_length=1):
     vecw = np.dot(M, vec)
     w = vecw[3]
-    # clip here.
     txs, tys, tzs = vecw[0] / w, vecw[1] / w, vecw[2] / w
-    tis = (0 <= vecw[0]) & (vecw[0] <= 1) & (0 <= vecw[1]) & (vecw[1] <= 1)
-    if np.any(tis):
-        tis = vecw[1] < 1
+    if np.isinf(focal_length):  # don't clip orthographic projection
+        tis = np.ones(txs.shape, dtype=bool)
+    else:
+        tis = ((-1 <= vecw[0]) & (vecw[0] <= 1)
+               & (-1 <= vecw[1]) & (vecw[1] <= 1)
+               & (tzs <= 0))
```

Comment:
Note that we may want to switch to `tzs <= -1` in the future, since a value of 0 means that there is no near clipping plane as in this image. But at the moment I thought that the view all the way up to the camera was most appropriate.

https://www.scratchapixel.com/lessons/3d-basic-rendering/perspective-and-orthographic-projection-matrix/opengl-perspective-projection-matrix.html
![image](https://github.com/matplotlib/matplotlib/assets/14363975/6ddfa8b8-89ec-44a3-9d1c-924b762b51a1)